### ISSUE TITLE
RN 0.59.8 - Unable to resolve module `pegjs`

### ISSUE THREAD

**Issue by flexaddicted** [react-native-community/react-native-svg#1007]
I'm running the latest version of RN (0.59.8) with the latest version of react-native-svg (9.4.0).
When I try to launch the application, both in iOS and Android, I receive the following error:

> bundling failed: Error: Unable to resolve module `pegjs` from `/Users/PATH_TO_APP/android/react-native-svg/lib/extract/extractTransform.js`: Module `pegjs` does not exist in the Haste module map

Do you know how I can fix this?

Thanks, Lorenzo

**Comment by msand** [comment]
The latest version has no dependencies at all. This error means you're loading old js in your app. Try clearing caches, reinstall the module from npm, reset the packager, and try again.

**Comment by kelset** [comment]
uhm... few thoughts:

1) please create a repro so that we can understand if it's actually something wrong with RN or the lib
2) maybe it's something cached wrongly?
```
rm -rf node_modules;
rm -rf $TMPDIR/react-native-packager-cache-*;
rm -rf $TMPDIR/metro-*;
rm -rf $TMPDIR/react-*;
rm -rf $TMPDIR/haste-*;
watchman watch-del-all;
yarn;
yarn start --reset-cache; (for starting bundler)
```

**Comment by flexaddicted** [comment]
@msand @kelset Thank you very much for your reply. I'll try and let you know.

**Comment by flexaddicted** [comment]
@msand @kelset I receive the same error. I've done all the steps described by Lorenzo.

To have more context, the update is from RN 0.57.8. I'm not sure if this one can cause problems.

I can try to recreate a repro.

**Comment by msand** [comment]
Try removing node_modules and running 
npm i react-native-svg
or
yarn add react-native-svg

**Comment by kelset** [comment]
Uhm then I'd suggest you use [rn-diff-repo](https://github.com/pvinis/rn-diff-purge) to double check that you did all the necessary steps/changes while upgrading.

**Comment by flexaddicted** [comment]
@msand @kelset I cannot find `pegjs` module in `node_modules`. I added in my `package.json` file as a dev dependency.

Now I receive the following:

> method.bind is not a function

**Comment by msand** [comment]
pegjs is not used by the latest version, you shouldn't need it, unless your node_modules are in a corrupted state.

**Comment by LuisRizo** [comment]
I'm not sure if this is related but after going from RN 0.57.8 to 0.59.4 I started getting the same `TypeError: method.bind is not a function`. The stack trace leads to Shape.js line 8

```
import { Component } from "react";
import SvgTouchableMixin from "../lib/SvgTouchableMixin";
import _ from "lodash";

class Shape extends Component {
    constructor() {
        super(...arguments);
        _.forEach(SvgTouchableMixin, (method, key) => {
            this[key] = method.bind(this); // <-- This guy
        });
        //noinspection JSUnusedGlobalSymbols
        this.state = this.touchableGetInitialState();
    }
}

export default Shape;
```

**Comment by flexaddicted** [comment]
@kelset @msand I've figured out what the problem was.

My Brownfield React Native project was including `react-native-svg` as a local dependency. In other words, I've downloaded the source code of `react-native-svg` and included it as a module served locally and not as a node module (this was due to fix some problems on the gradle file).

During the migration, Android Studio was building against that module and not the new one available in `node_modules`. That was causing the error.

I've fixed following this guide: https://stackoverflow.com/questions/23025433/how-to-clear-gradle-cache. In particular, I removed the gradle build cache.

Thanks for your support.
